pallets/xyk/src/lib.rs

```diff
@@ -453,15 +476,15 @@ decl_module! {
                 second_asset_reserve - second_asset_amount,
             );
 
-            if (first_asset_reserve - first_asset_amount == 0.saturated_into::<T::Balance>())
-                || (second_asset_reserve - second_asset_amount == 0.saturated_into::<T::Balance>()) {
+            if (first_asset_reserve - first_asset_amount == 0.saturated_into::<BalanceOf<T>>())
+                || (second_asset_reserve - second_asset_amount == 0.saturated_into::<BalanceOf<T>>()) {
```

Comment:
IMP!!
This may not be a part of modifications related to Tokens, but shouldn't LiquidityAssets and LiquidityPools also be removed?
Moreover, wouldn't zero balance in reserve cause divide by zero error in mint_liquidity function?
We could keep the pool alive by keeping LiquidityAssets and LiquidityPools, if we take measures to avoid divide by zero in the entire pallet. Which we should do any way as divide by zero might happen even if just one of the assets drops to zero.

Reply:
Also shouldn't mint_liquidity also use liquidity instead of first_asset_amount for symmetry?

Reply:
The idea was this:
If Pools are removed, the `mint_liquidity` is no longer callable for the pool. Instead, the pool can be recreated using `create_pool`. In this case, LiquidityAssets and LiquidityPools for the corresponding pool will be reinitialized with new values. This also means divizion by zero will not happen.

Reply:
as for using liquidity for minting- unfortunately we had to drop the idea of having mint and burn symmetrical because of the different semantics:
in both cases, the user calling the function is aiming to give some assets they possess in exchange for another asset. Unlike trading, is most cases they will be aiming to provide the exact amount of assets. Forcing them to provide a destination amount may result in unexpected amount to be taken from the user if some thirdparty transaction will affect the balance in the pool

Reply:
https://trello.com/c/SJ43cop3/105-figure-out-if-remove-liquiditypools-along-with-pools